### PR TITLE
Fix validator from crashing during test spec validation

### PR DESCRIPTION
**Pull request by dnkoutso**
closes https://github.com/CocoaPods/CocoaPods/issues/8484

Regressed on https://github.com/CocoaPods/CocoaPods/commit/2401391d9b1c278802542c8a4c8c56d492d832d3#diff-7f0b2bce7920a8edef5824debd63ffabR92 due to a missing test that exercises this logic.

### CODE REVIEW

**Comment by CocoaPodsBarista** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please include a CHANGELOG entry to credit yourself! <br />
You can find it at <a href="https://github.com/CocoaPods/CocoaPods/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
  </tbody>
</table>

Here's an example of your CHANGELOG entry:

```markdown
* Fix validator from crashing during test spec validation  
  [dnkoutso](https://github.com/dnkoutso)
  [#issue_number](https://github.com/CocoaPods/CocoaPods/issues/issue_number)
```

*note*: There are two invisible spaces after the entry's text.


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: lib/cocoapods/validator.rb [hunk 1]
```diff
@@ -584,8 +584,8 @@ def configure_pod_targets(targets, target_installation_results, deployment_targe
           (build_configuration.build_settings['OTHER_CFLAGS'] ||= '$(inherited)') << ' -Wincomplete-umbrella'
           build_configuration.build_settings['SWIFT_VERSION'] = (pod_target.swift_version || derived_swift_version) if pod_target.uses_swift?
         end
-        pod_target_installation_result.test_specs_by_native_target.each do |test_native_target, test_specs|
-          if pod_target.uses_swift_for_spec?(test_specs.first)
+        pod_target_installation_result.test_specs_by_native_target.each do |test_native_target, test_spec|
+          if pod_target.uses_swift_for_spec?(test_spec)
```
Comment (amorde) [inline lib/cocoapods/validator.rb]: is this always a 1 to 1 mapping? as in is there ever a scenario where there would be multiple test specs for a single target?

Comment (dnkoutso) [reply lib/cocoapods/validator.rb]: yes it is for non library specs (test specs and app specs)

**Comment by paulb777** [comment]
Is there a reasonable way to add the missing test?

**Comment by dnkoutso** [comment]
Also find out a bug that `--swift-version` is not honored because `pod_target#swift_version` no longer returns `nil`.

**Comment by dnkoutso** [comment]
Getting close to wrap this up. The tests inside `validator_spec.rb` are ..... 🤢

File: spec/unit/validator_spec.rb [hunk 2]
```diff
@@ -995,17 +998,31 @@ def test_swiftpod(swift_version = '3.1.0')
         Pod::Installer::PodSourceInstaller.any_instance.stubs(:unlock_files!)
         validator = Validator.new(file, config.sources_manager.master.map(&:url))
         validator.stubs(:build_pod)
-        validator.swift_version = swift_version
         validator.stubs(:validate_url)
         validator
       end
 
+      # Creates a test fixture using the `--swift-version` parameter.
+      def test_swiftpod_with_swift_version_parameter(swift_version = '3.1.0')
```
Comment (dnkoutso) [inline spec/unit/validator_spec.rb]: re factored these methods out to be explicit on what kind of setup a validator has with regards to Swift version.

File: lib/cocoapods/validator.rb [hunk 3]
```diff
@@ -628,8 +642,7 @@ def build_pod
       else
         UI.message "\nBuilding with `xcodebuild`.\n".yellow do
           scheme = if skip_import_validation?
-                     pod_target = @installer.pod_targets.find { |pt| pt.pod_name == spec.root.name }
-                     pod_target.label if pod_target.should_build?
+                     validation_pod_target.label if pod_target.should_build?
```
Comment (segiddins) [inline lib/cocoapods/validator.rb]: `if validation_pod_target.should_build?`

Comment (dnkoutso) [reply lib/cocoapods/validator.rb]: nice fixed.

File: lib/cocoapods/validator.rb (hunk 1 above)
Comment (dnkoutso) [reply lib/cocoapods/validator.rb]: This change right here is the actual crash fix.

**Comment by dnkoutso** [comment]
Build is green and have received approval. Verified test breaks without my changes.

**Pull request merged**